### ISSUE TITLE
FileEdit changed event value wrong

### ISSUE THREAD

**Issue by uschmidt83** [napari/magicgui#204]
This is a weird bug that happens when a file or folder is selected via the file dialog, but _not_ when the path is entered into `line_edit` or set via `.value`.

**To Reproduce**
Run
```python
from magicgui import magicgui
from pathlib import Path
@magicgui(p={'mode': 'd'})
def func(p: Path):
    pass
func.p.changed.connect(lambda event: print(event.value))
func.p.value = '/home'
func.show()
```
Click the "Choose directory" button and press "Open".

Output:
```
/home
FileEdit(mode='d', value=PosixPath('/home'))
```

**Expected behavior**
Output:
```
/home
/home
```
**Environment (please complete the following information):**
 - OS: Linux
 - backend: Qt 5.15.2
 - magicgui at a07b6d3885fb63a6d7947fbb03fc90a971759006

**Comment by tlambert03** [comment]
Hmm, that looks very similar to the bug I tried to fix in #173... thanks foe reporting

**Comment by uschmidt83** [comment]
Ah, I didn't know about #176. I tried to debug this myself, but gave up after a while...

**Comment by tlambert03** [comment]
much appreciated!  I'll try to take a look tomorrow.  I'll probably push another bug fix release soon, and will try to get #187 in before then as well.

**Comment by uschmidt83** [comment]
No worries, that's an easy bug to work around.
